File: closeUp/Classes/Reset.cpp
#include "Reset.h"
#include "HomeLayer.h"
#include "Letters.h"
#include "GameLayer.h"
Reset::Reset(void)
{
}

Reset::~Reset(void)
{
}
CCScene* Reset::scene()
{
	CCScene *scene=CCScene::create();
	CCLayer *layer=Reset::create();
	scene->addChild(layer);
	return scene;
}
bool Reset::init()
{
	if (!CCLayer::init())
	{
		return false;
	}

	CCSprite *bg=CCSprite::create("background.jpg");
	bg->setPosition(ccp(160,240));
	addChild(bg);

	//CCLayerColor *bai=CCLayerColor::create(ccc4(255,255,255,255),100,100);
	//bai->setPosition(ccp(160,200));
	//addChild(bai);

	CCMenuItemImage *back=CCMenuItemImage::create("back@2x.png","back-over@2x.png",this,menu_selector(Reset::backBtn));
	back->setScale(0.5);
	CCMenu *backMenu=CCMenu::create(back,NULL);
	backMenu->setPosition(ccp(44,459));
	addChild(backMenu);

	CCSprite *reset=Letters::create("green-button.png","Reset Game",30,ccc3(255,255,255));
	CCMenuItemSprite *reItem=CCMenuItemSprite::create(reset,reset,this,menu_selector(Reset::reSet));
	CCMenu *reMenu=CCMenu::create(reItem,NULL);
	reMenu->setPosition(ccp(160,30));
	CCMoveTo *to=CCMoveTo::create(0.4,ccp(160,330));
	reMenu->runAction(to);
	addChild(reMenu);


	return true;
}
void Reset::backBtn(CCObject *pSender)
{
	CCScene *scene=HomeLayer::scene();
	CCTransitionCrossFade *co=CCTransitionCrossFade::create(0.1,scene);
	CCDirector::sharedDirector()->replaceScene(co);
}
void Reset::reSet(CCObject *pSender)
{
	CCUserDefault::sharedUserDefault()->setIntegerForKey("LEVEL",0);
	CCUserDefault::sharedUserDefault()->setIntegerForKey("SCORE",0);
	CCUserDefault::sharedUserDefault()->setIntegerForKey("SKIPTIMES",2);
	CCUserDefault::sharedUserDefault()->setIntegerForKey("ZOOMTIMES",3);
	CCUserDefault::sharedUserDefault()->setIntegerForKey("REMOVETIMES",4);
	CCUserDefault::sharedUserDefault()->flush();
}

File: luaCU/script/Reset.lua
module("Reset",package.seeall)

function scene( ... )
	-- body
	local s=CCScene:create()
	layer=CCLayer:create()
	init()
	s:addChild(layer)
	return s
end
function init( ... )
	-- body
	local bg=CCSprite:create("background.jpg")
    bg:setScale(0.9)
    bg:setPosition(ccp(160,240))
    layer:addChild(bg)

	local backItem=CCMenuItemImage:create("back.png","back.png")
	backItem:registerScriptTapHandler(function ( ... )
		-- body
		AudioEngine.playEffect("sound/enter.wav")
		CCDirector:sharedDirector():replaceScene(Homelayer:scene())
	end)
	backItem:setScale(0.6)
	local backMenu=CCMenu:createWithItem(backItem)
	backMenu:setPosition(ccp(44,459))
	layer:addChild(backMenu)

	local resetItem=MyBtns:create("green-button.png","green-button-over.png" ,"Reset Games",40)
	resetItem:registerScriptTapHandler(function ( ... )
		-- body
		AudioEngine.playEffect("sound/enter.wav")
		CCUserDefault:sharedUserDefault():setIntegerForKey("LEVEL",0)
		CCUserDefault:sharedUserDefault():setIntegerForKey("SCORE",0)
		CCUserDefault:sharedUserDefault():setIntegerForKey("SKIPTIMES",2)
		CCUserDefault:sharedUserDefault():setIntegerForKey("ZOOMTIMES",5)
		CCUserDefault:sharedUserDefault():setIntegerForKey("REMOVETIMES",8)
		CCUserDefault:sharedUserDefault():flush()
	end)
	local reMenu=CCMenu:createWithItem(resetItem)
	reMenu:setPosition(ccp(160,30))
	local to=CCMoveTo:create(0.4,ccp(160,330))
	reMenu:runAction(to)
	layer:addChild(reMenu)

	local words1=CCLabelTTF:create("Thanks for playing Close Up Pics!","",15)
	words1:setPosition(ccp(160,260))
	layer:addChild(words1)
	local words2=CCLabelTTF:create("For support email","",15)
	words2:setPosition(ccp(160,240))
	layer:addChild(words2)
	local words3=CCLabelTTF:create("support@meidiaflex.co","",15)
	words3:setPosition(ccp(160,225))
	layer:addChild(words3)
	local words4=CCLabelTTF:create("Images are used on a Creative","",15)
	words4:setPosition(ccp(160,205))
	layer:addChild(words4)
end